packages/expect/src/jestMatchersObject.ts

```diff
@@ -22,8 +17,16 @@ const JEST_MATCHERS_OBJECT = Symbol.for('$$jest-matchers-object');
 // Jest may override the stack trace of Errors thrown by internal matchers.
 export const INTERNAL_MATCHER_FLAG = Symbol.for('$$jest-internal-matcher');
 
+export type JestRawMatcherFn = Expect.RawMatcherFn & {
+  [INTERNAL_MATCHER_FLAG]?: boolean;
+};
+
+export type JestMatchersObject = {
+  [matcherName: string]: JestRawMatcherFn;
+};
```

Comment:
These lived in `types.ts` file. Symbol was defined in this file, imported into `types` and typing were imported back. Somewhat circular. It is not possible to import symbol into `@jest/types`, because of circular dependancies. I moved these types here, because they are internal. Seemed like user should not know about them.